buildSrc/src/main/groovy/com/github/okbuilds/core/model/Target.groovy

```diff
@@ -39,31 +39,38 @@ abstract class Target {
     final Project project
     final Project rootProject
     final String name
+    final String testName
     final String identifier
     final String path
+
     final Set<String> sources = [] as Set
+    final Set<String> testSources = [] as Set
     final Set<Target> targetAptDeps = [] as Set
     final Set<Target> targetCompileDeps = [] as Set
+    final Set<Target> targetTestCompileDeps = [] as Set
 
     protected final Set<ExternalDependency> externalAptDeps = [] as Set
     protected final Set<ExternalDependency> externalCompileDeps = [] as Set
+    protected final Set<ExternalDependency> externalTestCompileDeps = [] as Set
 
     /**
      * Constructor.
      *
      * @param project The project.
      * @param name The target name.
      */
-    Target(Project project, String name) {
+    Target(Project project, String name, String testName = 'test') {
```

Comment:
this default should be in the java target. But lets leave it for now. will clean it up in a follow up